### PR TITLE
Fix getVersionedInput handler bugs and missing impl

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 1]
```diff
@@ -1338,6 +1339,25 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
...
+                        != request.getVersionedInputs().getRepositoryId()
+                    || !existingVersioningEntry
+                        .getCommit()
+                        .equals(request.getVersionedInputs().getCommit())) {
+                  throw new AlreadyExistsException(
+                      ModelDBConstants.DIFFERENT_REPOSITORY_OR_COMMIT_MESSAGE);
+                }
+              }
+            },
+            executor)
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:1360]: validation was missing to add here so added it.

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 2]
```diff
@@ -1347,11 +1367,13 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
...
-        .thenApply(
-            stringVersioningEntryMap -> stringVersioningEntryMap.get(request.getId()), executor);
+    final var futureVersionedInputs =
+        versionInputHandler.getVersionedInputs(Collections.singleton(request.getId()));
+    return privilegedVersionedInputsHandler
+        .filterVersionedInputsBasedOnPrivileges(
+            Collections.singleton(request.getId()),
+            futureVersionedInputs,
+            this::getEntityPermissionBasedOnResourceTypes)
+        .thenApply(versioningEntryMap -> versioningEntryMap.get(request.getId()), executor);
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:1377]: Added filter for versionedInputs based on Privileges

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java [hunk 3]
```diff
@@ -829,10 +829,14 @@ public void getVersionedInputs(
...
-                          .setVersionedInputs(versionedInputs)
-                          .build(),
+                  versionedInputs -> {
+                    GetVersionedInput.Response.Builder builder =
+                        GetVersionedInput.Response.newBuilder();
+                    if (versionedInputs != null) {
+                      builder.setVersionedInputs(versionedInputs);
+                    }
+                    return builder.build();
+                  },
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:839]: Fix null pointer if versionedInput not found as per old code

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java [hunk 4]
```diff
@@ -260,63 +261,92 @@ public VersionInputHandler(
...
+            .thenApply(
+                existingVersioningEntryMap -> existingVersioningEntryMap.get(entityId), executor);
+    return existingEntityFuture.thenCompose(
+        existingVersioningEntry -> {
+          if (existingVersioningEntry != null) {
+            if (existingVersioningEntry.getRepositoryId() != versioningEntry.getRepositoryId()
+                || !existingVersioningEntry.getCommit().equals(versioningEntry.getCommit())) {
+              return InternalFuture.failedStage(
+                  new AlreadyExistsException(
+                      ModelDBConstants.DIFFERENT_REPOSITORY_OR_COMMIT_MESSAGE));
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java:275]: Do not allowed to update from different repositories and commit (Logic from old code)

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java [hunk 5]
```diff
@@ -260,63 +261,92 @@ public VersionInputHandler(
...
+                    argsMaps.forEach(preparedBatch::add);
+                    int[] insertedCount = preparedBatch.execute();
+                    LOGGER.trace(
+                        "Inserted versioning_modeldb_entity_mapping count: "
+                            + insertedCount.length);
+                  }
+                }
+              });
+        },
+        executor);
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java:349]: Handled insert vs update properly with batch insertion

**Review by conradoverta — APPROVED** [review]

**Pull request merged**